### PR TITLE
refactor(material/button): switch to tokens theming API

### PR DESCRIPTION
**Pull request by wagnermaciel**

### CODE REVIEW

**Review by mmalerba — APPROVED** [review]
LGTM

File: src/material/core/tokens/m2/mdc/_button-filled.scss [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    focus-state-layer-color: $on-primary,
+    hover-state-layer-color: $on-primary,
+    label-text-color: if($on-color, $on-color, inherit),
+    pressed-state-layer-color: $on-primary
+  );
+}
+
+// Tokens that can be configured through Angular Material's typography theming API.
+@function get-typography-tokens($theme) {
+  @return ();
```
Comment (crisbeto) [inline src/material/core/tokens/m2/mdc/_button-filled.scss:75]: The typography and density styles are still using MDC's old API. Shouldn't we switch them over as well?

Comment (wagnermaciel) [reply src/material/core/tokens/m2/mdc/_button-filled.scss:75]: Yeah this is what I was trying to bring up in standup but didn't have time to dive into. Currently all button variants are relying on the same styles for density & typography. I asked if I should split those out and Andrew said he was ok with landing this without doing so

Comment (wagnermaciel) [reply src/material/core/tokens/m2/mdc/_button-filled.scss:75]: That said, I don't really know if I was able to explain the situation in the short time we had during standup. I'll ping Andrew and see if he can weigh in here

Comment (crisbeto) [reply src/material/core/tokens/m2/mdc/_button-filled.scss:75]: That's at least what we've been doing when converting the other components to tokens. For the filled button specifically it looks like we have these 5 tokens for the typography: `label-text-font`, `label-text-size`, `label-text-tracking`, `label-text-transform`, `label-text-weight` and the `container-height` token for the density.

Comment (andrewseguin) [reply src/material/core/tokens/m2/mdc/_button-filled.scss:75]: Had a chat with Wagner - it'll be a bit easier to land the typography changes across all the variants at once after these PRs land. It'll just be a different iterative approach that reaches the same goal in the end (remove the old MDC mixins)

**Review by andrewseguin — APPROVED** [review]

File: src/material/core/tokens/m2/mdc/_button-filled.scss (hunk 1 above)
Comment (crisbeto) [reply src/material/core/tokens/m2/mdc/_button-filled.scss:75]: Alright, I wanted to make sure that we do it before we consider the button as migrated.

**Pull request merged**